### ISSUE TITLE
Cannot uninstall Docker Desktop (hang)

### ISSUE THREAD

**Issue by CherryDT** [docker/for-win#13799]
### Description

I cannot uninstall Docker Desktop.

It hangs here:

![image](https://github.com/docker/for-win/assets/6010463/a4ea7a43-d1f2-4d73-aaca-fb02b6b4f29b)

`install-log-admin.txt` contains this:

```
-------------------------------------------------------------------------------->8
Version: 4.25.1 (128006)
Sha1: 
Started on: 2023.11.21 14:19:43.989
Resources: R:\Temp\DockerDesktop\4ep1tamhmid\resources
OS: Windows 10 Pro
Edition: Professional
Id: 2009
Build: 19045
BuildLabName: 19041.1.amd64fre.vb_release.191206-1406
File: C:\ProgramData\DockerDesktop\install-log-admin.txt
CommandLine: "R:\Temp\DockerDesktop\4ep1tamhmid\Docker Desktop Installer.exe" "uninstall" --remove-self
You can send feedback, including this log file, at https://github.com/docker/for-win/issues
[2023-11-21T14:19:44.234120200Z][ManifestAndExistingInstallationLoader][I] No install path specified, looking for default installation registry key
[2023-11-21T14:19:44.529812800Z][Installer][I] Existing installation found: build=128006, version=4.25.1, basePath=C:\Program Files\Docker\Docker, channelUrl=https://desktop-stage.docker.com/win/main/amd64/appcast.xml
[2023-11-21T14:19:45.124359900Z][UninstallWorkflow][I] Cancel pending background download
[2023-11-21T14:19:45.126375500Z][BackgroundTransfer][I] Cancel current background transfer job
[2023-11-21T14:19:45.135375900Z][UninstallWorkflow][I] Running before uninstall actions
[2023-11-21T14:19:45.135375900Z][UninstallWorkflow][I] Proceeding with CommunityInstaller.ExecAction
[2023-11-21T14:19:45.139377200Z][UninstallWorkflow-ExecAction][I] Running: C:\Program Files\Docker\Docker\InstallerCli.exe -k -u with timeout=60000
[2023-11-21T14:19:46.240512500Z][UninstallWorkflow][I] Done with CommunityInstaller.ExecAction
[2023-11-21T14:19:46.240512500Z][UninstallWorkflow][I] Proceeding with CommunityInstaller.KillLockingProcessesAction
[2023-11-21T14:19:47.753868300Z][UninstallWorkflow][E] Action CommunityInstaller.KillLockingProcessesAction failed: System.Exception: Could not list processes locking resources. Result code is 5
   bei CommunityInstaller.ResourceManager.GetLockingAppsAndServices()
   bei CommunityInstaller.KillLockingProcessesAction.<DoAsync>d__44.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bei CommunityInstaller.UninstallWorkflow.<ProcessAsync>d__13.MoveNext()
```

### Reproduce

Not sure how to get into this state

### Expected behavior

So apparently there is an error "Could not list processes locking resources. Result code is 5" but that should not cause a hang... it should probably just skip trying to list locked resources and proceed with the uninstall if this step fails.

### docker version

```bash
error during connect: in the default daemon configuration on Windows, the docker client must be run with elevated privileges to connect: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.24/version": open //./pipe
/docker_engine: The system cannot find the file specified.
Client:
 Cloud integration: v1.0.35+desktop.5
 Version:           24.0.6
 API version:       1.43
 Go version:        go1.20.7
 Git commit:        ed223bc
 Built:             Mon Sep  4 12:32:48 2023
 OS/Arch:           windows/amd64
 Context:           default
```

### docker info

```bash
Client:
 Version:    24.0.6
 Context:    default
 Debug Mode: false

Server:
ERROR: error during connect: in the default daemon configuration on Windows, the docker client must be run with elevated privileges to connect: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.24/info": open //./
pipe/docker_engine: The system cannot find the file specified.
errors pretty printing info
```

### Diagnostics ID

Cannot get it - it's stuck for 30 minutes showing "NoResponse GET /idle (541.9µs): Get "http://ipc/idle": open \\.\pipe\dockerBackendApiServer: The system cannot  find the file specified." every few seconds.

### Additional Info

Note: The `docker version` and `docker info` output says something about requiring admin privileges, but they were already run from an elevated command prompt.

**Comment by Yanksi** [comment]
Change the temporary directory back to default in RamDisk Configuration, and try to uninstall it again. This works for me.

**Comment by CherryDT** [comment]
Thanks it worked.

(Still, this would be a bug in my opinion.)

**Comment by alainbryden** [comment]
> Change the temporary directory back to default in RamDisk Configuration, and try to uninstall it again. This works for me.

For what it's worth, simply copying the file downloaded to `M:\Temp\DockerDesktopUpdates` to your Desktop (on C:) and re-running it from there (without having to re-mapping your temp folder) fixed the issue.

So it's less about having the temp folder remapped, and more a matter of running the updater not-from the C drive.

**Comment by aravindk777** [comment]
> temporary directory back to default in RamDisk Configuration, a

How do we change the temp directory path so that the uninstaller continue? Where do i change it? Right now, my uninstaller stuck as its been denied access at "c:\programdata\DockerDesktop\tmp-d4w" temp folder. I cant myself get rid of that folder too. Is there a way for the Uninstaller to get past this folder?

**Comment by Yanksi** [comment]
> > temporary directory back to default in RamDisk Configuration, a
> 
> How do we change the temp directory path so that the uninstaller continue? Where do i change it? Right now, my uninstaller stuck as its been denied access at "c:\programdata\DockerDesktop\tmp-d4w" temp folder. I cant myself get rid of that folder too. Is there a way for the Uninstaller to get past this folder?

Maybe you should try running the uninstaller as administrator?

**Comment by aravindk777** [comment]
Yes. But earlier attempts to run the Uninstaller as admin didn't work. So, for me at least, I had to restart my machine, log in as that admin account and run the Uninstaller from that account. That removed it.

**Comment by CherryDT** [comment]
> How do we change the temp directory path so that the uninstaller continue?

1. Open command prompt as admin
2. Enter `set TEMP=%LOCALAPPDATA%\Temp` (or whatever temp folder you want)
3. Enter `set TMP=%TEMP%`
4. Run the uninstaller from the command line

The new temp folder will be used only for programs launched from this command line session, so no need to set it back at the end.

**Comment by LupinIII** [comment]
Came here for the exact same problem (I even have my temp folder mapped to R:\ like in the original post). But seriously, Docker Desktop is the only program to ever have a problem with that. This should not happen.

**Comment by ordinaryCoder** [comment]
@aravindk777 
I tried changing the temp folder path as mentioned by @CherryDT  but it don't work for. 

Since my windows machine is a WVD I didn't had access to admin account but had the admin rights on my user account.

After manually deleting all the Docker Desktop references,

I Opened powershell  using "Run as Administrator" and took the ownership of the folder using below command

`takeown /F "C:\ProgramData\DockerDesktop\tmp-d4w" /A /R `

This allowed me to delete the folder "C:\ProgramData\DockerDesktop"

**Comment by birendra-k** [comment]
Thanks @ordinaryCoder.

**Comment by Universal-Invariant** [comment]
Same problem. I also had problems with previous versions when running docker. I haven't used docker in several months because of it but tried to update today and got the install error here leading me here. I'm now wondering if the run error which IIRC is reported as related to WSL. Supposedly it was fixed but the fix didn't not work. Also, when I first started using docker I had no issues. FYI, the last version I used was 4.42 and I just updated to 4.48. I've had some apps like Arduino have similar problems with temp files on ram drives.
